File: MaxTip.cpp
#include<algorithm>
using namespace std;
struct tip
{
    int a;
    int b;
    int d;
};
bool compared(tip x,tip y)
{
    return (x.d<y.d);
}
int main() {
	int t;
	cin>>t;
	while(t--)
	{
	    int n,x,y;
	    cin>>n>>x>>y;
	    int ar[n],br[n];
	    tip s[n];
	    for(int i=0;i<n;i++)
	    cin>>ar[i];
	    for(int i=0;i<n;i++)
	    cin>>br[i];
	    for(int i=0;i<n;i++)
	    {
	        s[i].a=ar[i];
	        s[i].b=br[i];
	        s[i].d=abs(s[i].a-s[i].b);
	    }
	    int sum=0;
	    sort(s,s+n,compared);
	    for(int i=n-1;i>=0;i--)
	    {
	        if(s[i].a>=s[i].b)
	        {
	            if(x>0)
	            {
	               sum+=s[i].a;
	               x--;
	            }
	            else
	            {
	                sum+=s[i].b;
	                y--;
	            }
	        }
	        else if(s[i].b>s[i].a)
	        {
	            if(y>0)
	            {
	               sum+=s[i].b;
	               y--;
	            }
	            else
	            {
	                sum+=s[i].a;
	                x--;
	            }
	        }
	    }
	    cout<<sum<<endl;
	}
	return 0;
}

File: MaxTip.java
import java.util.*;
class tip
{
	public int a;
	public int b;
	public int d;
	public tip(int a,int b,int d)
	{
		this.a=a;
		this.b=b;
		this.d=d;
	}
}
class diffsort implements Comparator<tip> 
{  
    public int compare(tip a, tip b) 
    { 
        return a.d - b.d; 
    } 
} 
class MaxTip
{
	public static void main(String[] args) {
		Scanner in=new Scanner(System.in);
		int t=in.nextInt();
		for(int z=0;z<t;z++)
		{
			int n,x,y;
			n=in.nextInt();
			x=in.nextInt();
			y=in.nextInt();
			int a[]=new int[n];
			int b[]=new int[n];
			for(int i=0;i<n;i++)
				a[i]=in.nextInt();
			for(int i=0;i<n;i++)
				b[i]=in.nextInt();
			ArrayList<tip> arr=new ArrayList<>();
			for(int i=0;i<n;i++)
			{
				arr.add(a[i],b[i],Math.abs(a[i]-b[i])); 
			}
			int sum=0;
			Collections.sort(arr,new diffsort());
			Collections.reverse(arr);
			for(int i=n-1;i>0;i--)
			{
				tip t=arr.get(i);
				if(t.a>t.b)
				{
					if(x>0)
					{
						sum+=t.a;
						x--;
					}
					else
					{
						sum+=t.b;
						y--;
					}
				}
				else
				{
					if(y>0)
					{
						sum+=t.b;
						y--;
					}
					else
					{
						sum+=t.a;
						x--;
					}
				}
			}

			System.out.println(sum);
		}
	}
}